x-pack/plugins/uptime/public/components/overview/alerts/monitor_expressions/time_expression_select.tsx

```diff
@@ -0,0 +1,128 @@
+import * as labels from '../translations';
+import { AlertFieldNumber } from '../alert_field_number';
+
+interface Props {
+  setAlertParams: (key: string, value: any) => void;
+}
+
+const TimeRangeOptions = [
+  {
+    'aria-label': labels.SECONDS_TIME_RANGE,
+    'data-test-subj': 'xpack.uptime.alerts.monitorStatus.timerangeUnitSelectable.secondsOption',
+    key: 's',
+    label: labels.SECONDS,
+  },
+  {
+    'aria-label': labels.MINUTES_TIME_RANGE,
+    'data-test-subj': 'xpack.uptime.alerts.monitorStatus.timerangeUnitSelectable.minutesOption',
+    checked: 'on',
+    key: 'm',
+    label: labels.MINUTES,
+  },
+  {
+    'aria-label': labels.HOURS_TIME_RANGE,
+    'data-test-subj': 'xpack.uptime.alerts.monitorStatus.timerangeUnitSelectable.hoursOption',
+    key: 'h',
+    label: labels.HOURS,
+  },
+  {
+    'aria-label': labels.DAYS_TIME_RANGE,
+    'data-test-subj': 'xpack.uptime.alerts.monitorStatus.timerangeUnitSelectable.daysOption',
+    key: 'd',
+    label: labels.DAYS,
+  },
+];
+
+export const TimeExpressionSelect: React.FC<Props> = ({ setAlertParams }) => {
+  const [numMins, setNumMins] = useState<number>(15);
+
+  const [timerangeUnitOptions, setTimerangeUnitOptions] = useState<any[]>(TimeRangeOptions);
+
+  useEffect(() => {
+    const timerangeUnit = timerangeUnitOptions.find(({ checked }) => checked === 'on')?.key ?? 'm';
+    setAlertParams('timerange', { from: `now-${numMins}${timerangeUnit}`, to: 'now' });
+  }, [numMins, timerangeUnitOptions, setAlertParams]);
+
+  return (
+    <EuiFlexGroup gutterSize="s">
+      <EuiFlexItem grow={false}>
+        <AlertExpressionPopover
+          aria-label={labels.OPEN_THE_POPOVER_TIME_RANGE_VALUE}
+          content={
+            <AlertFieldNumber
+              aria-label={labels.ENTER_NUMBER_OF_TIME_UNITS}
+              data-test-subj="xpack.uptime.alerts.monitorStatus.timerangeValueField"
+              disabled={false}
+              fieldValue={numMins}
+              setFieldValue={setNumMins}
+            />
+          }
+          data-test-subj="xpack.uptime.alerts.monitorStatus.timerangeValueExpression"
+          description="within"
+          id="timerange"
+          value={`last ${numMins}`}
+        />
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <AlertExpressionPopover
+          aria-label={i18n.translate(
+            'xpack.uptime.alerts.monitorStatus.timerangeUnitExpression.ariaLabel',
+            {
+              defaultMessage: 'Open the popover for time range unit select field',
+            }
+          )}
+          content={
+            <>
+              <EuiTitle size="xxs">
+                <h5>
+                  <FormattedMessage
+                    id="xpack.uptime.alerts.monitorStatus.timerangeSelectionHeader"
+                    defaultMessage="Select time range unit"
+                  />
+                </h5>
+              </EuiTitle>
+              <EuiSelectable
+                aria-label={i18n.translate(
```

Comment:
We could extract this to the translations file from which we imported the `labels`.